dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -331,12 +262,45 @@ winrt::guid Microsoft::Windows::AppNotifications::Helpers::RegisterComActivatorG
         THROW_IF_FAILED(hr);
     }
 
-    std::wstring notificationAppId{ RetrieveNotificationAppId() };
+    std::wstring notificationAppId{ Microsoft::Windows::AppNotifications::Helpers::RetrieveNotificationAppId() };
     RegisterAssets(notificationAppId, registeredGuid);
 
     // Remove braces around the guid string
     return winrt::guid(registeredGuid.substr(1, registeredGuid.size() - 2));
-    
+}
+
+void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring const& appId, std::wstring const& clsid)
+{
+    wil::unique_hkey hKey;
+    // subKey: \Software\Classes\AppUserModelId\{AppGUID}
+    std::wstring subKey{ c_appIdentifierPath + appId };
+
+    THROW_IF_WIN32_ERROR(RegCreateKeyEx(
+        HKEY_CURRENT_USER,
+        subKey.c_str(),
+        0,
+        nullptr /* lpClass */,
+        REG_OPTION_NON_VOLATILE,
+        KEY_ALL_ACCESS,
+        nullptr /* lpSecurityAttributes */,
+        &hKey,
+        nullptr /* lpdwDisposition */));
+
+    // Try the following techniques to retrieve display name and icon:
+    // 1. From the current process.
+    // 2. Based on the best app shortcut, using the FrameworkUdk.
+    // 3. Use the default assets.
+    Microsoft::Windows::AppNotifications::ShellLocalization::AppNotificationAssets assets{};
+
+    if (FAILED(Microsoft::Windows::AppNotifications::ShellLocalization::RetrieveAssetsFromShortcut(assets)) &&
+        FAILED(Microsoft::Windows::AppNotifications::ShellLocalization::RetrieveAssetsFromProcess(assets)))
+    {
+        THROW_IF_FAILED(Microsoft::Windows::AppNotifications::ShellLocalization::RetrieveDefaultAssets(assets));
+    }
+
+    RegisterValue(hKey, L"DisplayName", reinterpret_cast<const BYTE*>(assets.displayName.c_str()), REG_EXPAND_SZ, assets.displayName.size() * sizeof(wchar_t));
+    RegisterValue(hKey, L"IconUri", reinterpret_cast<const BYTE*>(assets.iconFilePath.c_str()), REG_EXPAND_SZ, assets.iconFilePath.size() * sizeof(wchar_t));
```

Comment:
We should check if the icon is empty. In the default case it is, so we probably shouldnt populate this registry.

Reply:
I have verified there are no issues if we write an empty string as the IconUri. There won't be an icon in the toast.

Let me verify if I can see the same behavior if we don't set IconUri at all.

Reply:
Well if the behavior is the same, I would keep it simple then and just have this as empty.

Reply:
No IconUri means that Shell will render the default icon 😊.

Reply:
Then you need to add a comment for that here. It isn't obvious.

Reply:
Yeah, of course. I added a comment right before writing to the registry.